Documentation/ceph-upgrade.md

```diff
@@ -319,6 +319,9 @@ step to upgrade the Prometheus RBAC resources as well.
 kubectl apply -f deploy/examples/monitoring/rbac.yaml
 ```
 
+Or, if you use only the `rook-ceph` operator Helm chart, you should also add `monitoring.enabled` to
```

Comment:
Do we also need to mention that setting this is not needed if they set the same thing in the cluster helm chart? It's in both places now, right?
```suggestion
Or, if you use only the `rook-ceph` operator Helm chart, you could enable monitoring by setting `monitoring.enabled`.
```

Reply:
I think this is needed in some fashion. The monitoring resources were previously at least partially applied by the Helm chart by default. I think it is actually a step users should do if they previously used Helm and monitoring. Maybe there is a better way I could word this?

Reply:
So if the operator and cluster are in the same namespace you have to specify it in the operator chart, and if in different namespaces you have to specify in the cluster chart? Seems like we would prefer it to be set in the cluster chart if they are using it, so maybe we should remove that condition.

Reply:
That's not quite right. 

To firstly clarify: nothing has changed about the behavior I mentioned in this PR. Currently, the `rook-ceph` chart is deployed to its namespace. The `rook-ceph-cluster` chart is deployed to a namespace and also takes an `operatorNamespace` value that tells the chart where the operator is located. If the `rook-ceph-cluster` chart is being deployed into the same namespace as `operatorNamespace`, it will assume the `rook-ceph` chart has already deployed the resources it needs (including the monitoring RBAC) and will not deploy them a second time. If we want to change that behavior, it is orthogonal to this PR.

Reply:
Ah thanks for the clarification, that would be a much bigger design change than just for this monitoring, so let's not touch it. So what wording do you recommend here? Seems like it should mention that if the cluster is created in the same namespace as the operator, the monitoring needs to be enabled with that setting.

Reply:
I can revise this wording, but I think it's generally safe to say that if you have monitoring enabled, set `monitoring.enabled` in the Helm chart at this point in the upgrade. 

Otherwise, we have to get deep into different cases of do or don't set the value:
- what if the user uses the `rook-ceph-cluster` chart inside the operator namespace (do set this value)
- what if the user uses the `rook-ceph-cluster` chart outside the operator namespace (don't set here)
   - but what if there is **also** a `rook-ceph-cluster` chart inside the operator namespace? (do still set here)
- what if the user creates cluster resources themselves in the operator namespace (do still set here)
- what if the user creates cluster resources themselves outside the operator namespace (don't set here)
   - but what if there is **also** a cluster inside the operator namespace (do still set here)
   
 4 out of 6 cases have the user setting this enabled

Reply:
Agreed, let's keep it simple. :)